connect-inject/container_init_test.go

```diff
@@ -142,6 +141,77 @@ consul-k8s connect-init -pod-name=${POD_NAME} -pod-namespace=${POD_NAMESPACE} \
 	}
 }
 
+func TestHandlerContainerInit_transparentProxy(t *testing.T) {
+	cases := map[string]struct {
+		globalEnabled     bool
+		annotationEnabled *bool
+		expectEnabled     bool
+	}{
+		"enabled globally, annotation not provided": {
+			true,
+			nil,
+			true,
+		},
+		"enabled globally, annotation is false": {
+			true,
+			pointerToBool(false),
+			false,
+		},
+		"enabled globally, annotation is true": {
+			true,
+			pointerToBool(true),
+			true,
+		},
+		"disabled globally, annotation not provided": {
+			false,
+			nil,
+			false,
+		},
+		"disabled globally, annotation is false": {
+			false,
+			pointerToBool(false),
+			false,
+		},
+		"disabled globally, annotation is true": {
+			false,
+			pointerToBool(true),
+			true,
+		},
+	}
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			h := Handler{EnableTransparentProxy: c.globalEnabled}
+			pod := minimal()
+			if c.annotationEnabled != nil {
+				pod.Annotations[annotationTransparentProxy] = strconv.FormatBool(*c.annotationEnabled)
+			}
+
+			expectedSecurityContext := &corev1.SecurityContext{
+				RunAsUser:  pointerToInt64(0),
+				RunAsGroup: pointerToInt64(0),
+				Capabilities: &corev1.Capabilities{
+					Add: []corev1.Capability{"NET_ADMIN"},
+				},
+			}
+			expectedCmd := `/consul/connect-inject/consul connect redirect-traffic \
+  -proxy-id="$(cat /consul/connect-inject/proxyid)" \
+  -proxy-uid=0`
+			container, err := h.containerInit(*pod, k8sNamespace)
+			require.NoError(t, err)
+
+			if c.expectEnabled {
+				require.Equal(t, expectedSecurityContext, container.SecurityContext)
+				cmd := strings.Join(container.Command, " ")
+				require.Contains(t, cmd, expectedCmd)
+			} else {
+				require.Nil(t, container.SecurityContext)
+				cmd := strings.Join(container.Command, " ")
+				require.NotContains(t, cmd, expectedCmd)
+			}
```

Comment:
minor refactor option

```suggestion
			require.NoError(t, err)
			actualCmd := strings.Join(container.Command, " ")

			if c.expectEnabled {
				require.Equal(t, expectedSecurityContext, container.SecurityContext)
				require.Contains(t, actualCmd, expectedCmd)
			} else {
				require.Nil(t, container.SecurityContext)
				require.NotContains(t, actualCmd, expectedCmd)
			}
```